subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -1376,7 +1384,7 @@ static ssize_t sendto_dtls_client(struct net_context *ctx, const void *buf,
 		}
 	}
 
-	if (!ctx->tls->tls_established) {
+	if (!is_handshake_complete(ctx)) {
 		/* TODO For simplicity, TLS handshake blocks the socket even for
```

Comment:
Ok, so it's actually documented that non-blocking TLS sockets aren't really non-blocking.